src/diffusers/loaders/lora_base.py

```diff
@@ -505,6 +504,12 @@ def _best_guess_weight_name(cls, *args, **kwargs):
         deprecate("_best_guess_weight_name", "0.35.0", deprecation_message)
         return _best_guess_weight_name(*args, **kwargs)
 
+    @property
+    def _merged_adapters(self):
+        if "_merged_adapters" not in self.__dict__:
+            self.__dict__["_merged_adapters"] = set()
+        return self.__dict__["_merged_adapters"]
```

Comment:
Since it's a `_merged_adapters` would have been class attribute otherwise, it would have side-effects. 

So,

```py
pipe = DiffusionPipeline(...)
print(pipe.fused_loras) # set()
pipe.unet.add_adapter(...)
pipe.fuse_lora(...)

pipe = DiffusionPipeline(...)
print(pipe.fused_loras) # {"default"}
```

We see that even for a new instantiation of the pipeline, `fused_loras` aren't automatically reset. This is undeseriable. The other solution would have been to always call `unfuse_lora()` after `fuse_lora()` but I didn't prefer that.

Reply:
We can it not be an instance attribute set during `__init__`?

Reply:
Mixin classes are not supposed to have an `__init__()`.

Reply:
Yeah we cannot really do that. Hence I had to resort to the current solution.

Reply:
I think a cleaner solution would be to re-assign the attribute instead of mutating it. That way, no `__dict__` shenanigans are required and there are no side effects:

```python
In [1]: class Foo:
   ...:     _merged_adapters = set()
   ...:     def fuse_lora(self, adapters):
   ...:         self._merged_adapters = self._merged_adapters | set(adapters)  # <= re-assign
   ...: 

In [2]: foo0 = Foo()

In [3]: foo0.fuse_lora(["foo", "bar"])

In [4]: foo0._merged_adapters
Out[4]: {'bar', 'foo'}

In [5]: foo1 = Foo()

In [6]: foo1._merged_adapters
Out[6]: set()

In [7]: foo1.fuse_lora(["bar", "baz"])

In [8]: foo1._merged_adapters
Out[8]: {'bar', 'baz'}

In [9]: foo0._merged_adapters
Out[9]: {'bar', 'foo'}
```

Reply:
Do the recent changes work for you? 

If not, I would appreciate either a direct PR to this branch or suggestions that work in the full context of this PR.